<?xml version="1.0"?>
<launch>
    <!-- spawn Upper controller  !! -->
    <node pkg="asoc_upper_controller" type="upper_controller_anti_rec" name="upper_controller" clear_params="true" output="screen">
        <param name = "controller_freq" type="double" value = "40.0"/>
        <param name = "goalRadius" type="double" value = "0.5"/>
        <param name = "goal_pose_err" type="double" value = "1.0"/>

        <param name = "baseSpeed" type="double" value = "5.0"/>
        <param name = "para_vel" type="double" value = "5.0"/>
        <param name = "rot_angle" type="double" value = "0.0"/>
        <param name = "P_Long" type="double" value = "35.0"/>
        <param name = "I_Long" type="double" value = "0.0"/>
        <param name = "D_Long" type="double" value = "2.0"/>

        <param name = "P_Yaw" type="double" value = "27.0"/>
        <param name = "I_Yaw" type="double" value = "0.0"/>
        <param name = "D_Yaw" type="double" value = "25.0"/>
        <param name = "forward_dist" type="double" value = "0.1"/>
        <param name = "forward_dist2" type="double" value = "0.4"/>
        

        <param name = "P_Lateral" type="double" value = "20.0"/>
        <param name = "I_Lateral" type="double" value = "1.0"/>
        <param name = "D_Lateral" type="double" value = "20.0"/>

        <param name = "Kp" type="double" value = "1.0"/>
        <param name = "Kd" type="double" value = "0.0"/>
        <param name = "zero_pos" type="double" value = "1.0"/>
        <param name = "roll_rot_factor" type="double" value = "0.8"/>        
        <param name = "roll_lat_factor" type="double" value = "1.0"/>
        <param name = "velocity_factor" type="double" value = "0"/>
        <param name = "P_pit" type="double" value = "200.0"/>
        <param name = "P_rol" type="double" value = "200.0"/>
        <param name = "D_pit" type="double" value = "1.0"/>
        <param name = "D_rol" type="double" value = "1.0"/>
        <param name = "slow_ff" type="double" value = "1.0"/>
            </node>
</launch>

